pkg/k8shandler/collection.go

```diff
@@ -0,0 +1,166 @@
+  /*err := sdk.Create(collectionPriorityClass)
+  if err != nil && !errors.IsAlreadyExists(err) {
+    logrus.Fatalf("Failure creating Collection priority class: %v", err)
+  }*/
+
+  return nil
+}
+
+func createOrUpdateFluentdServiceAccount(logging *logging.ClusterLogging) error {
+
+  fluentdServiceAccount := utils.ServiceAccount("aggregated-logging-fluentd", logging.Namespace)
+
+  err := sdk.Create(fluentdServiceAccount)
+  if err != nil && !errors.IsAlreadyExists(err) {
+    logrus.Fatalf("Failure creating Fluentd service account: %v", err)
+  }
+
+  return nil
+}
+
+func createOrUpdateFluentdConfigMap(logging *logging.ClusterLogging) error {
+
+  // TODO: update to use the actual files we want...
+  fluentdConfigMap := utils.ConfigMap(
+    "logging-fluentd",
+    logging.Namespace,
+    map[string]string{
+      "fluent.conf": string(utils.GetFileContents("files/fluent.conf")),
+      "throttle-config.yaml": string(utils.GetFileContents("files/fluentd-throttle-config.yaml")),
+      "secure-forward.conf": string(utils.GetFileContents("files/secure-forward.conf")),
+    },
+  )
+
+  err := sdk.Create(fluentdConfigMap)
+  if err != nil && !errors.IsAlreadyExists(err) {
+    logrus.Fatalf("Failure constructing Fluentd configmap: %v", err)
+  }
+
+  return nil
+}
+
+func createOrUpdateFluentdSecret(logging *logging.ClusterLogging) error {
+
+  fluentdSecret := utils.Secret(
+    "logging-fluentd",
+    logging.Namespace,
+    map[string][]byte{
+      "ca": utils.GetFileContents("/tmp/_working_dir/ca.crt"),
+      "key": utils.GetFileContents("/tmp/_working_dir/system.logging.fluentd.key"),
+      "cert": utils.GetFileContents("/tmp/_working_dir/system.logging.fluentd.crt"),
+      "ops-ca": utils.GetFileContents("/tmp/_working_dir/ca.crt"),
+      "ops-key": utils.GetFileContents("/tmp/_working_dir/system.logging.fluentd.key"),
+      "ops-cert": utils.GetFileContents("/tmp/_working_dir/system.logging.fluentd.crt"),
+    }  )
+
+  err := sdk.Create(fluentdSecret)
+  if err != nil && !errors.IsAlreadyExists(err) {
+    logrus.Fatalf("Failure constructing Fluentd secret: %v", err)
+  }
+
+  return nil
+}
+
+func createOrUpdateFluentdDaemonset(logging *logging.ClusterLogging) error {
+
+  fluentdPodSpec := fluentdPodSpec(logging)
+  fluentdDaemonset := utils.DaemonSet("logging-fluentd", logging.Namespace, "fluentd", "fluentd", fluentdPodSpec)
+
+  err := sdk.Create(fluentdDaemonset)
+  if err != nil && !errors.IsAlreadyExists(err) {
+    logrus.Fatalf("Failure creating Fluentd Daemonset %v", err)
+  }
+
+  return nil
+}
+
+func fluentdPodSpec(logging *logging.ClusterLogging) v1.PodSpec {
+  return v1.PodSpec{
+    NodeSelector: map[string]string {
+      "logging-infra-fluentd": "true",
+    },
+    PriorityClassName: "cluster-logging",
+    Containers: []v1.Container{
+      {Name: "fluentd", Image: "openshift/origin-logging-fluentd:latest", ImagePullPolicy: v1.PullIfNotPresent,
```

Comment:
Maybe not this PR, but doesnt this need to come from a manifest... or possibly someplace we can override?

Reply:
this is addressed a bit more in #11